### PR TITLE
Perf Improvement in XmlJsonWriter.WriteEscapedJsonString.

### LINKED ISSUE #7579: DataContractJsonSerializer in CoreClr is slower than Desktop in string scenarios

### PR DESCRIPTION
**Pull request by shmao**
PR #6665 touched a very hot code path, which caused some Perf regressions. This PR is to address the perf issue (basically redid the original fix). After this fix, the perf result would be better than the result prior to #6665.

Fix #7579.

### CODE REVIEW

**Comment by shmao** [comment]
@SGuyGe @khdang

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX please

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX please

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX please

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test

**Comment by shmao** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlJsonWriter.cs [hunk 1]
```diff
@@ -71,6 +74,26 @@ public XmlJsonWriter(bool indent, string indentChars)
...
+                }
+                else
+                {
+                    escapedJsonString = string.Format(CultureInfo.InvariantCulture, "\\u{0:x4}", ch);
+                }
+                
+                s_escapedJsonStringTables[ch] = escapedJsonString;
+            }
+        }
+
```
Comment (stephentoub) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlJsonWriter.cs]: Nit: it'd be slightly better if this didn't use an explicit static cctor, e.g.
```C#
private static readonly string[] s_escapedJsonStringTables = CreateEscapedJsonStringTables();

private static string[] CreateEscapedJsonStringTables()
{
    var tables = new string[WHITESPACE];
    for (int ch = 0; ch < WHITESPACE; ch++)
    {
        ch abbrev;
        tables[ch] = TryEscapeControCharacter((char)ch, out abbrev) ?
            string.Concat(BACK_SLASH, abbrev) :
            string.Format(CultureInfo.InvariantCulture, "\\u{0:x4}", ch);
    }
    return tables;
}
```

Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlJsonWriter.cs]: I've removed the static ctor per the suggestion. Thanks.

**Comment by shmao** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Comment by khdang** [comment]
LGTM

**Comment by SGuyGe** [comment]
LGTM

**Pull request merged**